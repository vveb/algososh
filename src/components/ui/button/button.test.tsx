import renderer from 'react-test-renderer';

import { Button } from './button';
import { fireEvent, render, screen } from '@testing-library/react';

describe('Button rendering', () => {

  it('Button with text rendering successfull', () => {
    const tree = renderer
    .create(<Button text='Some text'/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Button without text rendering successfull', () => {
    const tree = renderer
    .create(<Button />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Button disabled rendering successfull', () => {
    const tree = renderer
    .create(<Button disabled={true} />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Button with loader rendering successfull', () => {
    const tree = renderer
    .create(<Button isLoader={true} />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('some', () => {
  it('Button has correct onClick event processing', () => {
    window.alert = jest.fn();
    render(<Button onClick={() => alert('Button has clicked')}/>)
    const button = screen.getByRole('button');
    fireEvent.click(button);
    expect(window.alert).toHaveBeenCalledWith('Button has clicked');
  });
});
