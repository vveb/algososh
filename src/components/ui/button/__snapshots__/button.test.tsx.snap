// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button rendering should render Button component disabled 1`] = `
<button
  className="text text_type_button text_color_primary undefined undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button rendering should render Button component with loader 1`] = `
<button
  className="text text_type_button text_color_primary undefined undefined undefined "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    src="test-file-stub"
  />
</button>
`;

exports[`Button rendering should render Button component with text 1`] = `
<button
  className="text text_type_button text_color_primary undefined undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Some text
  </p>
</button>
`;

exports[`Button rendering should render Button component without text 1`] = `
<button
  className="text text_type_button text_color_primary undefined undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
